rust/crd/src/affinity.rs

```diff
@@ -62,7 +61,9 @@ mod test {
                   cleaner: true
         "#;
 
-        let history: SparkHistoryServer = serde_yaml::from_str(input).expect("illegal test input");
+        let deserializer = serde_yaml::Deserializer::from_str(input);
+        let history: SparkHistoryServer =
+            serde_yaml::with::singleton_map_recursive::deserialize(deserializer).unwrap();
```

Comment:
what sorcery is this ?

Reply:
serde_yaml expects use to use yaml tags when we have a complex enum (which logFileDirectory is).
We need to tell it to stop that
See https://docs.rs/serde_yaml/0.9.25/serde_yaml/with/singleton_map_recursive/index.html